app/models/katello/glue/pulp/repo.rb

```diff
@@ -757,8 +757,38 @@ def capsule_download_policy
         end
       end
 
+      def distributors_change?(capsule_distributors)
+        generated_distributors = self.generate_distributors.map { |dist| object_to_hash(dist) }
+        capsule_distributors.each do |dist|
+          dist.merge!(dist["config"])
+          dist.delete("config")
+        end
+
+        generated_distributors.any? do |gen_dist|
+          capsule_distributors.any? do |cap_dist|
+            (gen_dist.to_a - cap_dist.to_a).empty?
+          end
+        end
+      end
+
+      def importer_changed?(capsule_importer)
+        generated_importer = self.generate_importer(true)
+        download_policy = generated_importer.respond_to?(:download_policy) ? generated_importer.download_policy : nil
+        generated_importer_config = {  "download_policy": download_policy || nil,
+                                       "remove_missing": generated_importer.remove_missing || nil,
+                                       "feed": generated_importer.feed || nil
+                                    }.delete_if { |_k, v| v.nil? }.to_json
```

Comment:
download_policy and remove_missing are fairly yum-centric, does this work for non-yum importers?